Sources/StreamChat/StateLayer/ChatClient+Factory.swift

```diff
@@ -36,7 +36,10 @@ extension ChatClient {
     /// - Returns: An instance of ``ChannelList`` which represents actions and the current state of the list.
     public func makeChannelList(with query: ChannelListQuery, dynamicFilter: ((ChatChannel) -> Bool)? = nil) async throws -> ChannelList {
         let channels = try await channelListUpdater.update(channelListQuery: query)
-        return ChannelList(channels: channels, query: query, dynamicFilter: dynamicFilter, channelListUpdater: channelListUpdater, client: self)
+        let channelList = ChannelList(channels: channels, query: query, dynamicFilter: dynamicFilter, channelListUpdater: channelListUpdater, client: self)
+        let state = channelList.state
+        syncRepository.trackChannelListQuery { [weak state] in state?.query }
+        return channelList
```

Comment:
`ChannelList` is a struct, that is why we use state instead which is a class.

Reply:
Should it be a struct, though? 🤔  Maybe a point to add to the Tech Discussion, but I don't think that Chat, ChannelList, MemberList, etc... should be structs. For these reasons:
- They contain reference types, so there's not much value on them to be value types
- Since there's no value on them being a struct (Unless I'm missing something), it is not worth the additional binary size that Structs produce, more details here: https://forums.swift.org/t/reducing-value-witness-size-in-struct-composition/52172
- It will be difficult to mock for testing (For us and for the customers), which would require exposing protocols instead of the structs directly.

Reply:
Currently `Chat` and all the `XYZState` are classes, other types are structs. `Chat` could also be a struct. These are currently structs since they do not hold any mutable state. 
Currently there is no strong reason of `non-State` types must be structs. Let's discuss this in the tech meeting.

Reply:
Discussion result: let's change them to classes in a separate commit/PR

Reply:
This is done in the base branch.